BackEnd/src/business/drafting.ts

```diff
@@ -0,0 +1,198 @@
+import { Server } from 'socket.io';
+import NodeCache from 'node-cache';
+import { draftStepConfig } from '../../../Common/constants';
+import { DraftState } from '../../../Common/Interface/Internal/drafting';
+import { MakeId } from '../../../Common/utils';
+import logger from '../../logger';
+import { createDbDraft, getDbDraft } from '../db/drafting';
+
+const gameCache = new NodeCache();
+const gameTimers: Record<string, number[]> = {};
+
+const DEFAULT_DRAFT_TIME = 30;
+
+// 0 is spectator, 1 is blue, 2 is red
+function getTeam(game: DraftState, auth: string): 'blueTeam' | 'redTeam' {
+  if (game.blueTeam.auth === auth) {
+    return 'blueTeam';
+  }
+  else if (game.redTeam.auth === auth) {
+    return 'redTeam';
+  }
+  return null;
+}
+
+function clearTimersForRoom(room: string) {
+  while(gameTimers[room].length) {
+    clearInterval(gameTimers[room].pop());
+  }
+}
+
+export function createDraft(blueName: string, redName: string): DraftState {
+  const draftState: DraftState = {
+    blueTeam: {
+      ready: false,
+      bans: Array(5).fill('0'),
+      picks: Array(5).fill('0'),
+      disabledPicks: [],
+      auth: MakeId(10),
+      displayName: blueName,
+    },
+    redTeam: {
+      ready: false,
+      bans: Array(5).fill('0'),
+      picks: Array(5).fill('0'),
+      disabledPicks: [],
+      auth: MakeId(10),
+      displayName: redName,
+    },
+    stage: 0,
+    roomId: MakeId(10),
+    timerMax: DEFAULT_DRAFT_TIME,
+    timerRemaining: DEFAULT_DRAFT_TIME,
+    roomActive: false,
+  };
+  gameCache.set(draftState.roomId, draftState, 60 * 60 * 12); // Keep drafts in memory for 12 hours
```

Comment:
What happens if a draft is in proccess when we deploy or something happens to the server, is it lost to the void?

Reply:
Yeah it's lost in the void.